src/app/services/client/course.ts

```diff
@@ -0,0 +1,89 @@
+import { db } from "../../../../firebase";
+import { Course } from "@interfaces/db";
+import { courseConverter } from "../firestore";
+import {
+  deleteDoc,
+  doc,
+  DocumentReference,
+  PartialWithFieldValue,
+  setDoc,
+  updateDoc,
+} from "firebase/firestore";
+
+type addCourse = Pick<Course, "name" | "location" | "professors"> &
+PartialWithFieldValue<Course>
+
+export const addCourse = async (
+  course: addCourse,
+  courseID: string,
+) => {
```

Comment:
Shouldn't we let Firestore handle unique courseid generation instead? I think we can take advantage of the addDoc here instead of the setDoc